packages/ui/src/lib/layouts/Page.spec.ts

```diff
@@ -147,3 +148,29 @@ test('Expect subtitle is defined and cut', async () => {
   // expect class has the clamp
   expect(subtitleElement).toHaveClass('line-clamp-1');
 });
+
+test('Expect disconected observer after onDestroy', async () => {
+  const subtitle = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit';
+  const mutationObserverMock = vi.fn().mockImplementation(() => {
+    return {
+      observe: vi.fn().mockImplementation(() => vi.fn()),
+      disconnect: vi.fn(),
+      takeRecords: vi.fn(),
+    };
+  });
+
+  (global as any).MutationObserver = mutationObserverMock;
+
+  const page = render(Page, {
+    title: '',
+    subtitle,
+  });
+
+  const subtitleElement = screen.getByText(subtitle);
+  expect(subtitleElement).toBeInTheDocument();
+  expect(mutationObserverMock.mock.results[0].value.observe).toHaveBeenCalledTimes(1);
+  expect(mutationObserverMock.mock.results[0].value.disconnect).toHaveBeenCalledTimes(0);
+  page.unmount();
+  expect(mutationObserverMock.mock.results[0].value.disconnect).toHaveBeenCalledTimes(1);
+  expect(mutationObserverMock.mock.results[0].value.disconnect).not.toHaveBeenCalledTimes(2);
```

Comment:
```suggestion
test('Expect disconected observer after onDestroy', async () => {
  const subtitle = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit';

  const page = render(Page, {
    title: '',
    subtitle,
  });

  const subtitleElement = screen.getByText(subtitle);
  expect(subtitleElement).toBeInTheDocument();
  expect(MutationObserver.prototype.observe).toHaveBeenCalledTimes(1);
  expect(MutationObserver.prototype.disconnect).toHaveBeenCalledTimes(0);
  // run unmount to trigger the onDestroy callback
  page.unmount();
  expect(MutationObserver.prototype.disconnect).toHaveBeenCalledTimes(1);
});
```

I would avoid changing the global object without restoring it.

You could add at the begining of the test

```ts
let mutationObserverObserve: typeof MutationObserver.prototype.observe | undefined;
let mutationObserverDisconnect: typeof MutationObserver.prototype.disconnect | undefined;

beforeAll(() => {
  mutationObserverObserve = MutationObserver.prototype.observe;
  mutationObserverDisconnect = MutationObserver.prototype.disconnect;

  MutationObserver.prototype.observe = vi.fn();
  MutationObserver.prototype.disconnect = vi.fn();
});

afterAll(() => {
  if(!mutationObserverObserve || !mutationObserverDisconnect) {
    return;
  }
  MutationObserver.prototype.observe = mutationObserverObserve;
  MutationObserver.prototype.disconnect = mutationObserverDisconnect;
});
```